include/CAPI.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "ClientConfig.h"
+#include "Client.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+   struct c_client;
+   struct c_procedure;
+   struct c_invocation_response;
+
+   struct c_client* c_create_client(const char* usrname, const char* pwd, const char* hostname,
+         unsigned short port, bool keepConnecting, /*voltdb::ClientAuthHashScheme schema,*/
+         bool enableAbandon, bool enableTimeout, int timeoutInSec, bool useSSL);
+
+   void c_close(struct c_client* config);
+
+   struct c_procedure* c_create_call();
+   void c_drop_procedure(c_procedure*);
+
+   struct c_invocation_response* c_exec_proc(struct c_client* client, struct c_procedure* proc, const char** params);
+   struct c_invocation_response* c_exec_adhoc(struct c_client* client, struct c_procedure* proc, const char* param);
+   int c_status_code(struct c_invocation_response*);
+   void c_destroy_response(struct c_invocation_response*);
+
+   struct c_stringified_table {
```

Comment:
It looks like the client has no way of know what data type each column is and just would have to deal with the string representation of the column which is not very flexible

Reply:
Right. Supporting all column types would be some extra work, especially with memory management for the C++ world (and all the typings).
I created a follow-up ticket ENG-15044 to track this.